src/Spice86.Core/Emulator/CPU/VectorTableManager.cs

```diff
@@ -0,0 +1,38 @@
+namespace Spice86.Core.Emulator.CPU;
+
+using Spice86.Core.Emulator.Memory;
+
+/// <summary>
+/// Manages accesses to the vector table
+/// </summary>
+public class VectorTableManager {
+    private readonly Memory _memory;
+
+    public VectorTableManager(Memory memory) {
+        _memory = memory;
+    }
+    
+    /// <summary>
+    /// Installs an interrupt vector in the vector table.
+    /// </summary>
+    /// <param name="vectorNumber">The interrupt vector number to install.</param>
+    /// <param name="segment">The segment address of the vector.</param>
+    /// <param name="offset">The offset address of the vector.</param>
+    public void InstallVectorInTable(byte vectorNumber, ushort segment, ushort offset) {
+        uint offsetAddress = (uint)(4 * vectorNumber);
+        
+        // install the vector in the vector table
+        _memory.UInt16[offsetAddress] = offset;
+        _memory.UInt16[offsetAddress + 2] = segment;
+    }
+
+    /// <summary>
+    /// Reads the vector table at the given index (vectorNumber) and returns the address of the handler
+    /// </summary>
+    /// <param name="vectorNumber">vector to read</param>
+    /// <returns>A pair of ushort values that are segment and offset</returns>
+    public (ushort segment, ushort offset) ReadVectorFromTable(byte vectorNumber) {
```

Comment:
This functionality is also duplicated in `Spice86.Core.Emulator.InterruptHandlers.VGA.VgaFunctionality.GetInterruptVectorAddress` and `Spice86.Core.Emulator.ReverseEngineer.CSharpOverrideHelper.InterruptCall`

Reply:
Nice catch! This means this class was even more needed.
I refactored it to use an indexer and tuples.